src/main/java/com/coveo/spillway/SpillwayFactory.java

```diff
@@ -56,6 +56,20 @@ public SpillwayFactory(LimitUsageStorage storage, Clock clock) {
     this.clock = clock;
   }
 
+  /**
+   * Creates a new {@link Spillway} for a single limit
+   *
+   * @param <T> The type of the context. String if not using a propertyExtractor
+   *            ({@link LimitBuilder#of(String, java.util.function.Function)}).
+   *
+   * @param resource The name of the resource on which the limit are enforced
+   * @param limit The enforced limit
+   * @return The new {@link Spillway}
+   */
+  public final <T> Spillway<T> enforce(String resource, Limit<T> limit) {
```

Comment:
Wouldn't it be better to remove the throw and make the `enforce` put the limit in a set before creating the spillway? i.e. we do not throw, but we also do not enforce limits with the same name.

Reply:
I discussed this with @dreisch-coveo, I like it, but it means a breaking change in the API

Reply:
Introducing the exception was already a breaking change no? And the version published is alpha.

Reply:
yeah... its kind of sketchy that the only version available for use is alpha yet we have a bunch of applications pretty much using it for production =P

Reply:
If you want to do a safer breaking change, then make the API explicitly only accept a Set, remove the list/varargs method -- force the callers to consciously deal with it.

Reply:
Yeah I think that would be the cleanest way to do that. +1

Reply:
After further discussion with @dreisch-coveo, this is actually problematic since a Set doesn't enforce the same ordering as an Array/List.

2 approaches: 
1) leave this as is 
2) change the exception to a runtime exception

Reply:
TBH I'm not sure how much / whether ordering matters in terms of how the stuff is applied, but its a scary change to make without really digging deep into it.

#2 still forces the users to deal with the change if Java complains sufficiently about catching checked exceptions that aren't obviously thrown (I forget...)

Personally I think #1 or #2 are both fine

Reply:
Lots of `Set` implementations keep the order (ImmutableSet, SortedSet, TreeSet,..)
The order might or might not be important to users.

I'm personally a fan of non-silent and fail-fast approaches. Using a Set in the method signature doesn't necessarily makes it more explicit that two Limits with the same name will be `equal`.

For what it's worth, I prefer solution 2.

Also, if this is not an alpha product anymore, please release it under a non-alpha version.